lib/licensed/sources/nuget.rb

```diff
@@ -0,0 +1,230 @@
+# frozen_string_literal: true
+require "json"
+require "reverse_markdown"
+
+module Licensed
+  module Sources
+    # Only supports ProjectReference (project.assets.json) style restore used in .NET Core.
+    # Does not currently support packages.config style restore.
+    class NuGet < Source
+      def self.type
+        "nuget"
+      end
+
+      class NuGetDependency < Licensed::Dependency
+        LICENSE_FILE_REGEX = /<license\s*type\s*=\s*\"\s*file\s*\"\s*>\s*(.*)\s*<\/license>/ix.freeze
+        LICENSE_URL_REGEX = /<licenseUrl>\s*(.*)\s*<\/licenseUrl>/ix.freeze
+        PROJECT_URL_REGEX = /<projectUrl>\s*(.*)\s*<\/projectUrl>/ix.freeze
+
+        def initialize(name:, version:, path:, search_root: nil, metadata: {}, errors: [])
+          super(name: name, version: version, path: path, search_root: search_root, metadata: metadata, errors: errors)
+          @metadata["homepage"] = project_url if project_url
+        end
+
+        def nuspec_path
+          name = @metadata["name"]
+          File.join(self.path, "#{name}.nuspec")
+        end
+
+        def nuspec_contents
+          return unless nuspec_path
+          @nuspec_contents ||= File.read(nuspec_path)
+        end
+
+        def project_url
+          return unless nuspec_contents
+          @project_url ||= begin
+            match = nuspec_contents.match PROJECT_URL_REGEX
+            match[1] if match && match[1]
+          end
+        end
+
+        def project_files
+          @nuget_project_files ||= begin
+            files = [super(), nuspec_local_license_file].flatten.compact
```

Comment:
If a local license file referenced from the nuspec has a standard filename, e.g. `LICENSE`, this will recognize that file twice.  WDYT about only adding `nuspec_local_license_file` if `files` is either empty or only contains `Licensee::ProjectFiles::PackageManagerFile` objects?

Reply:
It feels a bit weird to me since it means we're potentially ignoring the actual license file according to NuGet and choosing a different one... but that's admittedly pretty unlikely.

Any reason not to just check that `files` doesn't already contain that filepath?

Reply:
Made a change to simply check that `files` doesn't already have a file with that name, let me know what you think of that.

Reply:
Resolving per our discussion since this approach seems reasonable.